test/e2e/job.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/apis/experimental"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// How long to wait for a job to finish.
+	jobTimeout = 5 * time.Minute
+
+	// Job selector name
+	jobSelectorKey = "job"
+)
+
+var _ = Describe("Job", func() {
+	f := NewFramework("job")
+	parallelism := 2
+	completions := 8
+	lotsOfFailures := 20 // considerably more than completions
+
+	// Simplest case: all pods succeed promtly
+	It("should run a job to completion when tasks succeed", func() {
+		By("Creating a job")
+		job := newTestJob("succeed", "all-succeed", api.RestartPolicyNever, parallelism, completions)
+		job, err := createJob(f.Client, f.Namespace.Name, job)
+		Expect(err).NotTo(HaveOccurred())
+		// Cleanup jobs when we are done.
+		defer func() {
```

Comment:
This is not strictly necessary. The Framework deletes the namespace and the namespace controller should then cleanup these jobs.